framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -2568,8 +2584,118 @@ protected boolean isTypeCastSafe(AnnotatedTypeMirror castType, AnnotatedTypeMirr
             }
         }
 
+        // Check when casting the expression having type T (a type variable) to type T.
+        // As the compiler will not report the unchecked warning in this case, so we should
+        // do the check for our type system when the subtype relation of the instantiations of the
+        // two T cannot be statically verified.
+        // See CastFromTtoT.java:22 for an example.
```

Comment:
Line number references are brittle. Can you refer to a method within that class?

Reply:
I refined the comment, I think it's good to redirect to that file now.